### PR TITLE
Miscellaneous fixes

### PR DESCRIPTION
**Pull request by vincent-botbol**
This PR addresses the remarks made in https://github.com/CatalaLang/catala/issues/709

### CODE REVIEW

File: tests/en/test_nsw_art_union.catala_en.result [hunk 1]
```diff
@@ -10,7 +10,9 @@ declaration scope Test1:
   my_gaming scope GamingAuthorized
 
 scope Test1:
-  definition my_gaming.artUnion equals
+  definition
```
Comment (denismerigoux) [inline tests/en/test_nsw_art_union.catala_en.result:13]: it should not split `definition my_gaming.artUnion equals` into three lines if `definition my_gaming.artUnion equals` does not overflow over 80 characters of wideness

File: tests/fr/droit_successions.catala_fr.result [hunk 2]
```diff
@@ -135,7 +135,8 @@ champ d'application RéserveHéréditaire:
   définition patrimoine_assiette_réserve_héréditaire égal à
     patrimoine_total
     - (
-      somme argent de (
+      somme argent
```
Comment (denismerigoux) [inline tests/fr/droit_successions.catala_fr.result:138]: this is weird ? The previous version was better ?

**Review by denismerigoux — APPROVED** [review]
Thanks! Now it seems that lines like `definition x equals` are automatically broken into three lines?

File: tests/en/test_nsw_art_union.catala_en.result (hunk 1 above)
Comment (vincent-botbol) [reply tests/en/test_nsw_art_union.catala_en.result:13]: It depends on the input, in this particular case the input was :
```
scope Test1:
  definition
    my_gaming.artUnion
  equals
    ArtUnion {
      -- grossProceeds: $50000
      -- typeOrg: NonProfit
      -- proceedsToBenefitingOrg: $25000
      -- totalValueOfThePrizes: $45000
      -- maxAmountMoneyAsSeparatePrize: $100
      -- holdsAuthority: true
    }
  assertion my_gaming.authorized = true
```
Hence, there is a separation happening.

More generally, if `( definition ... equals )` is on several lines, then it splits in three.

File: tests/fr/droit_successions.catala_fr.result (hunk 2 above)
Comment (vincent-botbol) [reply tests/fr/droit_successions.catala_fr.result:138]: Yes, this is a corner case with parenthesized expression. For this case, I tested manually on your input as I didn't find much instances of this in the code I was able to crawl.

Comment (denismerigoux) [reply tests/fr/droit_successions.catala_fr.result:138]: I guess it's fine, the new version is not that bad either.

Comment (vincent-botbol) [reply tests/fr/droit_successions.catala_fr.result:138]: I'll try and add an exception for this particular case.

File: tests/en/test_nsw_art_union.catala_en.result (hunk 1 above)
Comment (denismerigoux) [reply tests/en/test_nsw_art_union.catala_en.result:13]: Consider the whole definition syntax : `definition [x] equals [e]` with `[x]` and `[e]` being placeholder for potentially big strings/expressions.

For me, `definition [x] equals` on one hand, and `[e]` on the other hand, should be treated separately as to deciding whether to split or not a one-line component into several lines. So on this example, I would still expect `definition my_gaming.artUnion equals` to not split and be on one line, even if there is a big expression taking multiple lines after the `equals`.

Comment (vincent-botbol) [reply tests/en/test_nsw_art_union.catala_en.result:13]: That's exactly what it does actually, it looks at [definition x equals] as one specific block checks whether it contains some line-breaks and split in three or leave it as a one-liner.
For instance:
```
définition x égal à 1+ 2
définition y égal à
1+ 2
```
stays together or splits after the "égal à"
```
définition x égal à 1 + 2
définition x égal à 
  1 + 2
```

What you would want require topiary to be able to determine if it would fit or not. And that's the main issue, it's not able to do that so it relies on the existing user line breaks.

Comment (denismerigoux) [reply tests/en/test_nsw_art_union.catala_en.result:13]: OK we're on the same page. What I still don't understand is why this PR splits the line `definition my_gaming.artUnion equals` at the place where I tagged it. It was on a single line before, thus it should remain on a single line? Or have you manually inserted line breaks at these places?

Comment (vincent-botbol) [reply tests/en/test_nsw_art_union.catala_en.result:13]: Ah ok, I get your misunderstanding.
The original input is what I quoted:
```
  definition
    my_gaming.artUnion
  equals
    ...
```
The previously committed test result (which means the formatted original input)  was:
``` 
definition my_gaming.artUnion equals
    ...
```
BUT the changes contains rules that consider break-lines in such constructions and therefore the new committed result is with line-breaks which is similar to the original.

Sorry for the quid pro quo, it seemed obvious to me.

File: tests/fr/droit_successions.catala_fr.result (hunk 2 above)
Comment (vincent-botbol) [reply tests/fr/droit_successions.catala_fr.result:138]: I cannot manage to express this case.. I'll leave it as is for now
